### PR TITLE
Remove not needed includes in server side "Deathmatch" project

### PR DESCRIPTION
**Pull request by CrosRoad95**
As in title, i went through most of server side deathmatch project and removed what possible. refactor is minimal just to decrease amount of code, includes, not needed includes.

### CODE REVIEW

**Comment by tederis** [comment]
Headers like CSphere.h or CBox.h are ubiquitous and I think it would be better to put them into PCH.

**Comment by CrosRoad95** [comment]
box is used in ~3 files, sphere in ~10, meanwhile CVector is used everywhere

File: Server/mods/deathmatch/logic/CResourceMapItem.h [hunk 1]
```diff
@@ -10,8 +10,6 @@
  *****************************************************************************/
 #pragma once
 
-#include "CResourceFile.h"
-
```
Comment (botder) [inline Server/mods/deathmatch/logic/CResourceMapItem.h:14]: The class in this file inherits from that base class. Why do you want to remove the include for it here?

File: Server/mods/deathmatch/logic/packets/CPedWastedPacket.h [hunk 2]
```diff
@@ -12,7 +12,6 @@
 #pragma once
 
 #include "CPacket.h"
-#include <CVector.h>
```
Comment (botder) [inline Server/mods/deathmatch/logic/packets/CPedWastedPacket.h:15]: Are you removing the include because it's already in the precompiled header?

Comment (CrosRoad95) [reply Server/mods/deathmatch/logic/packets/CPedWastedPacket.h:15]: yes

File: Server/mods/deathmatch/logic/CResourceMapItem.h (hunk 1 above)
Comment (CrosRoad95) [reply Server/mods/deathmatch/logic/CResourceMapItem.h:14]: it compiles, i suspect it is included in different file

Comment (botder) [reply Server/mods/deathmatch/logic/CResourceMapItem.h:14]: IMHO a header must include everything it needs to "stand on its own". A precompiled header is just a tool to help with compilation times and, again my own opinion, headers should not be written in a style where such precompiled header exists.

Comment (CrosRoad95) [reply Server/mods/deathmatch/logic/CResourceMapItem.h:14]: i didn't say it coming from stdinc.h

Comment (botder) [reply Server/mods/deathmatch/logic/CResourceMapItem.h:14]: Yes, but that's not the problem I see with it.

Comment (CrosRoad95) [reply Server/mods/deathmatch/logic/CResourceMapItem.h:14]: what's the problem? how can i solve it?

**Pull request closed without merging**